h2o-core/src/main/java/water/Paxos.java

```diff
@@ -36,9 +36,15 @@
   // This is a packet announcing what Cloud this Node thinks is the current
   // Cloud, plus other status bits
   static synchronized int doHeartbeat( H2ONode h2o ) {
+    
```

Comment:
@michalkurka This is additional change to this PR. 

I realised that we should first check if we even need to consider the request ( based on cloud_name_hash equality ).

Otherwise a node from different cluster ( with different jar_hash) could kill this cluster.